src/flow/src/adapter.rs

```diff
@@ -120,10 +119,10 @@ pub struct FlowWorkerManager {
     /// which is `!Send` so a handle is used
     pub worker_handles: Vec<Mutex<WorkerHandle>>,
     /// The query engine that will be used to parse the query and convert it to a dataflow plan
-    query_engine: Arc<dyn QueryEngine>,
+    pub query_engine: Arc<dyn QueryEngine>,
     /// Getting table name and table schema from table info manager
     table_info_source: TableSource,
-    frontend_invoker: RwLock<Option<Box<dyn FrontendInvoker + Send + Sync>>>,
+    frontend_invoker: RwLock<Option<FrontendInvoker>>,
```

Comment:
We should be able to avoid using locks. When constructing FlowWorkerManager, passing in frontend_invoker should suffice.

Reply:
There is a depend when building standalone:
```
flownode -> node_manager -> ddl_task_executor -> FrontendInvoker
```
if not use mutex, then there will be:
```
FrontendInvoker -> flownode
```
which will be cyclic

> We should be able to avoid using locks. When constructing FlowWorkerManager, passing in frontend_invoker should suffice.